### PR TITLE
feat(sdb): Return  `[]byte{0x01}` for `GetCode(..)` of precompile contracts

### PR DESCRIPTION
**Pull request by itsdevbear**
### Summary by CodeRabbit

- New Feature: Introduced a `PrecompilePlugin` interface and integrated it into various components such as the state database, miner, and host setup. This allows for more flexible handling of precompiled contracts.
- Refactor: Updated the initialization of `precompile.NewPlugin` and `state.NewPlugin` instances in the `host.go` file to ensure correct setup based on the current block's rules.
- Test: Added generated mock packages for testing purposes, including mocks for `Accesslist`, `Log`, `Refund`, `SelfDestructStatePlugin`, `SelfDestructs`, `TransientStorage`, `Plugin`, and `PrecompilePlugin`.
- Chore: Updated `.mockery.yaml` configuration settings for two packages to include all files recursively and set the `with-expecter` flag to true.

### CODE REVIEW

File: eth/core/processor.go [hunk 1]
```diff
@@ -87,12 +87,6 @@ func NewStateProcessor(
 		statedb:  statedb,
 	}
 
-	if sp.pp == nil {
-		sp.pp = precompile.NewDefaultPlugin()
-	} else {
-		sp.BuildAndRegisterPrecompiles(sp.pp.GetPrecompiles(nil))
-	}
```
Comment (calbera) [inline eth/core/processor.go:94]: still need this

File: eth/core/processor.go [hunk 2]
```diff
@@ -124,6 +118,7 @@ func (sp *StateProcessor) Prepare(evm *vm.GethEVM, header *types.Header) {
 	// *technically* the precompiles change based on the chain config rules, to be fully correct,
 	// we should check every block.
 	sp.BuildAndRegisterPrecompiles(precompile.GetDefaultPrecompiles(&rules))
+	sp.BuildAndRegisterPrecompiles(sp.pp.GetPrecompiles(&rules))
```
Comment (calbera) [inline eth/core/processor.go:121]: only need to re-register the geth stateless precompiles every block

File: eth/core/processor.go (hunk 1 above)
Comment (itsdevbear) [reply eth/core/processor.go:94]: moved it to the miner.

File: eth/core/processor.go (hunk 2 above)
Comment (itsdevbear) [reply eth/core/processor.go:121]: will be needed (at least for now), in my geth tx pool branch

**Review by calbera — APPROVED** [review]

**Pull request merged**